csclient/csclient.go

```diff
@@ -99,6 +100,11 @@ func (c *Client) DisableStats() {
 	c.statsDisabled = true
 }
 
+// SetHeader sets custom header to be sent to the charm store on each request.
+func (c *Client) SetHeader(header http.Header) {
```

Comment:
SetHTTPHeader ?
(the charmstore client isn't really about HTTP, so it's perhaps better to be more specific here)